pkg/reconciler/events/scheduler/scheduler.go

```diff
@@ -95,7 +94,7 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, scheduler *v1.CloudSched
 
 func (r *Reconciler) reconcileJob(ctx context.Context, scheduler *v1.CloudSchedulerSource, topic, jobName string) error {
 	if scheduler.Status.ProjectID == "" {
-		projectID, err := utils.ProjectID(scheduler.Spec.Project, metadataClient.NewDefaultMetadataClient())
+		projectID, err := utils.ProjectIDOrDefault(ctx, scheduler.Spec.Project)
```

Comment:
Is the logic correct that Source also read projectID from env? I remember at first, we didn't have such logic for sources?

Reply:
You are right. This PR is to solve #1799 which is to use environment variable in sources. That's why I want to have one single function to do it. The motivation is that sources should have the same logic as brokers in terms of resolving project id. Please check the sources part to see if you are seeing any place that we shouldn't. The issue comes out from my previous discussion with Nacho and Grant.